src/Symfony/Component/Messenger/Transport/TransportFactory.php

```diff
@@ -53,6 +53,11 @@ public function createTransport(#[\SensitiveParameter] string $dsn, array $optio
             $packageSuggestion = ' Run "composer require symfony/beanstalkd-messenger" to install Beanstalkd transport.';
         }
 
+        $scheme = $this->parseScheme($dsn);
+        if ($scheme) {
+            throw new InvalidArgumentException(\sprintf('No transport supports the given Messenger DSN (with scheme "%s").', $scheme).$packageSuggestion);
```

Comment:
it would be better to show the DSN minus the authorization credentials - I created this PR because I often end up having to debug what the DSN was, maybe showing `amqp://username:******@host/foobar` for example?